Whisky/Views/Bottle/Pins/PinCreationView.swift

```diff
@@ -0,0 +1,124 @@
+                    panel.canCreateDirectories = false
+                    panel.begin { result in
+                        if result == .OK {
+                            if let url = panel.urls.first {
+                                let oldUrl = newPinURL.lastPathComponent
+                                                      .replacingOccurrences(of: ".exe", with: "")
+                                                      .replacingOccurrences(of: ".msi", with: "")
+                                                      .replacingOccurrences(of: ".bat", with: "")
+                                newPinURL = url
+                                // Only reset newPinName if the textbox hasn't been modified
+                                if newPinName.isEmpty || newPinName == oldUrl {
+                                    newPinName = url.lastPathComponent
+                                                    .replacingOccurrences(of: ".exe", with: "")
+                                                    .replacingOccurrences(of: ".msi", with: "")
+                                                    .replacingOccurrences(of: ".bat", with: "")
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            let pinPath = newPinURL.prettyPath()
+            Text(pinPath)
+                .truncationMode(.middle)
+                .lineLimit(2, reservesSpace: true)
+                .help(pinPath)
+            HStack {
+                Button("create.cancel") {
+                    dismiss()
+                }
+                .keyboardShortcut(.cancelAction)
+                .frame(maxWidth: .infinity, alignment: .trailing)
+                Button("pin.create") {
+                    let newlyCreatedPin = Program(name: newPinName, url: newPinURL, bottle: bottle)
+                    let existingProgram = bottle.programs.first(where: { program in
+                        program.url == newlyCreatedPin.url
+                    })
+                    // Ensure this URL isn't already pinned
+                    isDuplicate = existingProgram != nil && existingProgram?.pinned ?? false
+                    if !isDuplicate {
+                        // If this is a new program, add it to the array
+                        if existingProgram != nil {
+                            bottle.programs.append(newlyCreatedPin)
+                        }
+                        newlyCreatedPin.pinned = true
+                        // Trigger a reload
+                        bottle.settings.pins = bottle.settings.pins
+                        bottle.updateInstalledPrograms()
+                        dismiss()
+                    }
+                }
+                .keyboardShortcut(.defaultAction)
+                .disabled(newPinName.isEmpty || newPinURL == bottle.url.appending(path: "drive_c"))
+                .alert(String(localized: "pin.error.title"),
```

Comment:
Don't need the `String(localized: ..`.  `.alert("pin.error.title",...` should do the trick here

Reply:
Agreed, got carried away with explicit localization. Fixed in latest commit.